### ISSUE TITLE
vue/component-tags-order is never applied

### ISSUE THREAD

**Issue by matthew-dean** [vuejs/eslint-plugin-vue#2103]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.32.0
- **eslint-plugin-vue version:** 9.9.0
- **Node version:** 16.19.0
- **Operating System:** macOS

**Please show your full configuration:**

```json5
const vueOverrides = {
  /**
   * I wanted to turn this on, but it appears this rule only works with SFCs,
   * and most of this codebase doesn't use SFCs.
   */
  // 'vue/no-undef-components': ['error', {
  //   ignorePatterns: ['RouterLink', 'RouterView']
  // }],
  /**
   * Aligns with the Vue style guide.
   * @see https://v2.vuejs.org/v2/style-guide/?redirect=true#Component-name-casing-in-templates-strongly-recommended
   */
  'vue/component-name-in-template-casing': [
    'error',
    'PascalCase',
    {
      registeredComponentsOnly: false
    }
  ],

  /**
   * This is a good guard for JavaScript, but with TypeScript, we can allow
   * optional properties and use optional chaining to narrow values.
   */
  'vue/require-default-prop': 0,

  /**
   * Existing components don't conform to this,
   * and, in this case, it seems reasonable to turn off
   * since components are in PascalCase.
   */
  'vue/multi-word-component-names': 0,

  /** Sensible rules for HTML vs custom components */
  'vue/html-self-closing': ['error', {
    html: {
      void: 'any',
      normal: 'never',
      component: 'always'
    }
  }],
  'vue/component-tags-order': ['error', {
    order: [
      // ['script[setup]', 'template', 'style']
      ['style', 'template', 'script']
    ]
  }]
}

module.exports = {
  root: true,
  env: {
    node: true
  },
  extends: ['plugin:vue/vue3-strongly-recommended', 'standard', 'plugin:storybook/recommended'],
  parser: 'vue-eslint-parser',
  parserOptions: {
    parser: '@typescript-eslint/parser',
    extraFileExtensions: ['.vue']
  },
  rules: {
    /**
     * Okay tryyyying not to bikeshed here but I've never Standard JS's
     * formatting preference for space before in use before using StandardJS
     */
    'space-before-function-paren': ['error', {
      anonymous: 'never',
      named: 'never',
      asyncArrow: 'always'
    }],
    ...vueOverrides
  },
  overrides: [
    {
      files: ['*.ts', '.tsx', '.vue'],
      plugins: ['@typescript-eslint'],
      extends: [
        'plugin:vue/vue3-strongly-recommended',
        'standard-with-typescript'
      ],
      parserOptions: {
        parser: '@typescript-eslint/parser',
        project: './tsconfig.json',
        extraFileExtensions: ['.vue']
      },
      rules: {
        '@typescript-eslint/space-before-function-paren': ['error', {
          anonymous: 'never',
          named: 'never',
          asyncArrow: 'always'
        }],
        /**
         * Some methods return a promise as a convenience, like the router,
         * but it's not necessary to always prepend "void"
         */
        '@typescript-eslint/no-floating-promises': ['warn', {
          ignoreVoid: true
        }],
        /**
         * This needs to be a warning because it's in conflict with TypeScript's
         * decorator behavior.
         *
         * @see: https://github.com/typescript-eslint/typescript-eslint/issues/546
         */
        '@typescript-eslint/consistent-type-imports': ['warn', {
          prefer: 'type-imports'
        }],
        /**
         * This is a dangerous one to change based on an error,
         * because it's unclear what the original intent was.
         * However, we should warn so we avoid in the future.
         */
        '@typescript-eslint/prefer-nullish-coalescing': 'warn',
        /**
         * Loosen some rules to not force as much code refactoring
         * See individual rules for what issues these can cause.
         */
        '@typescript-eslint/strict-boolean-expressions': 0,
        '@typescript-eslint/restrict-template-expressions': 0,
        '@typescript-eslint/method-signature-style': 0,
        '@typescript-eslint/explicit-function-return-type': 0,
        ...vueOverrides
      }
    },
    /** Test eslinting rules */
    {
      files: ['test/**/*.ts', '**/__tests__/*.ts', '**/*.test.ts'],
      rules: {
        '@typescript-eslint/no-non-null-assertion': 0
      }
    },
    /** Storybook eslinting rules */
    {
      files: ['*.stories.ts'],
      rules: {
        '@typescript-eslint/consistent-type-assertions': 0
      }
    }
  ]
}

```

**What did you do?**	

```vue
<script></script>
<style></style>
<template>
  <div></div>
</template>

```

**What did you expect to happen?**
So, I know this is not a configuration issue, because every other ESLint / eslint-plugin-vue rule applies just fine, both from CLI and within VSCode. However, this rule is never applied. It wasn't applied before, even though docs say it has been included in `recommended` settings (and I'm using `strongly-recommended`), and it doesn't apply when defining it explicitly as an override.

No matter what, ESLint + eslint-plugin-vue says the order of the tags is fine, no matter in which way / order they violate this rule.

**What actually happened?**

The raw output is essentially nothing, because they pass.

**Repository to reproduce this issue**

I tried setting this up with CodeSandbox, but I can't figure out how to get ESLint to run at all. Is there a repo I can clone?

**Comment by ota-meshi** [comment]
> I tried setting this up with CodeSandbox, but I can't figure out how to get ESLint to run at all. Is there a repo I can clone?

Please share your repository with us.

**Comment by PieterBoeren** [comment]
@ota-meshi: I also see this issue, and I can also reproduce it in the [online playground](https://ota-meshi.github.io/eslint-plugin-vue-demo/), just open it with the default options, enable the "Priority C: Recommended" rules and I would expect an linting issue (since the default order for the tags is script => template => style and de default file in the playground has a different order).

**Comment by ota-meshi** [comment]
Please share a link that can reproduce the issue.
